From 65699cd73f4561c99a02c62bcdcab472dd4f7f9f Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 13 Feb 2025 11:20:51 +0800
Subject: [PATCH 090/104] mtk: mt76: mt7996: fix multi-band ACS issue

Fix missing assignment. Without this patch the following statement in
mt76_scan_complete() will not be True,

if (dev->scan.mlink == (struct mt76_vif_link *)vif->drv_priv)
        dev->drv->vif_link_remove(phy, vif, NULL, dev->scan.mlink);

which messes up the band setting on bss_info.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 scan.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scan.c b/scan.c
index eb823065..b0c523ab 100644
--- a/scan.c
+++ b/scan.c
@@ -205,6 +205,7 @@ int mt76_hw_scan(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 				 * initialized and assigned to a chanctx.
 				 * Here we use the default link to perform scan.
 				 */
+				mlink = (struct mt76_vif_link *)vif->drv_priv;
 				memcpy(&vif->bss_conf, link_conf, sizeof(struct ieee80211_bss_conf));
 				ret = dev->drv->vif_link_add(phy, vif, &vif->bss_conf, NULL);
 				if (ret)
-- 
2.45.2

